src/client/Client.js

```diff
@@ -492,6 +569,34 @@ class Client extends BaseClient {
     if (typeof options.retryLimit !== 'number' || isNaN(options.retryLimit)) {
       throw new TypeError('CLIENT_INVALID_OPTION', 'retryLimit', 'a number');
     }
+    if (typeof x !== 'undefined') {
```

Comment:
`x` isn't being defined anywhere else that I can see.

Reply:
:facepalm: I copied the first bit, from Sugden's post and forgot to change the variable :D This is the type of thing TS would've caught like immediately... oh well, I fixed it